docs/content/blog/authelia-traefik-setup-guide/index.md

```diff
@@ -0,0 +1,374 @@
+  - Authelia is on a different host to the proxy
+- All services are part of the `{{< sitevar name="domain" nojs="example.com" >}}` domain:
+- This domain and the subdomains will have to be adapted in all examples to match your specific domains unless you're
+  just testing, or you want to use that specific domain
+
+## File Structure
+The first thing we want to do is set up the file structure. Which should look something like this:
+```text
+📁 project
+ ┣ 📁 authelia
+ ┃  ┣ 📁 config
+ ┃  ┃ ┣ 📄 configuration.yaml
+ ┃  ┃ ┗ 📄 users.yaml
+ ┃  ┗ 📁 secrets
+ ┣ 📄 compose.yaml
+ ┗ 📁 traefik
+    ┣ 📁 config
+    ┃ ┣ 📄 dynamic.yaml
+    ┃ ┗ 📄 traefik.yaml
+    ┣ 📁 data
+    ┃ ┗ 📄 acme.json
+    ┣ 📁 logs
+    ┗ 📁 secrets
+```
+
+## Traefik and Whoami
+{{< callout context="note" title="Note" icon="outline/info-circle" >}}
+We'll focus on the minimal configuration needed to work with Authelia. For advanced Traefik features and configurations, consult their documentation.
+{{< /callout >}}
+
+Next, we'll set up Traefik as our reverse proxy. For detailed Traefik documentation, refer to the [official Traefik docs](https://doc.traefik.io/traefik/).
+
+#### Docker Compose
+
+
+```yaml{title="compose.yaml"}
+services:
+  traefik:
+    image: traefik:latest
+    container_name: traefik
+    restart: unless-stopped
+    security_opt:
+      - no-new-privileges=true
+    networks:
+      proxy: {}
+      authelia:
+        aliases:
+          - '{{< sitevar name="subdomain-authelia" nojs="auth" >}}.{{< sitevar name="domain" nojs="example.com" >}}'
+    ports:
+      - '80:80'
+      - '443:443'
+    environment:
+      TZ: 'America/Los_Angeles' # see below
+      CF_EMAIL_FILE: '/run/secrets/cloudflare_email'
+      CF_DNS_API_TOKEN_FILE: '/run/secrets/cloudflare_api_key'
+    volumes:
+      - '/var/run/docker.sock:/var/run/docker.sock:ro'
+      - './traefik/config/traefik.yaml:/traefik.yaml:ro'
+      - './traefik/config/dynamic.yaml:/dynamic.yaml:ro'
+      - './traefik/data/:/data'
+      - './traefik/logs:/logs'
+    secrets:
+      - cloudflare_email
+      - cloudflare_api_key
+    labels:
+      traefik.enable: 'true'
+      traefik.http.routers.dashboard.rule: 'Host(`traefik.{{< sitevar name="domain" nojs="example.com" >}})'
+      traefik.http.routers.dashboard.entrypoints: 'https'
+      traefik.http.routers.dashboard.middlewares: 'authelia@docker'
+      traefik.http.routers.dashboard.service: 'api@internal'
+
+  whoami:
+    image: traefik/whoami
+    restart: unless-stopped
+    container_name: whoami
+    labels:
+      traefik.enable: 'true'
+      traefik.http.routers.whoami.rule: 'Host(`whoami.{{< sitevar name="domain" nojs="example.com" >}}`)'
+      traefik.http.routers.whoami.entrypoints: 'https'
+    networks:
+      proxy: {}
+
+secrets:
+  cloudflare_email:
+    file: ./traefik/secrets/cloudflare_email.txt
+  cloudflare_api_key:
+    file: ./traefik/secrets/cloudflare_api_key.txt
+
+networks:
+  proxy:
+    external: true
```

Comment:
There is no mention of this network in the guide. If this should be the primary proxy net, authelia itself has no access to it, which will generate the following traefik warning:
```
2024-12-23T22:41:06+01:00 WRN Defaulting to first available network (&{"project_authelia" "172.19.0.2" '\x00' "" "7ce3d349ad84c0cb0be75dd9aacf8da0977850a380c134b887452243a9051186"}) for container "/authelia". container=authelia-project-6baa555da2ba674e94ebc96d0e11ed733645e77842ef03f87b1872c19029d962 providerName=docker serviceName=authelia-project
```
A solution would be to either remove the `proxy` network entirely or adding authelia to the network as well.

Reply:
While yes its not mentioned, which I will fix, Authelia does not need to be on/have access to the `proxy` network to work correctly. This is an ideal situation because it prevents malicious services on the `proxy` network from inspecting non-tls traffic from authelia

Reply:
Yeah, sorry. I didn't think of that. Setting the `traefik.docker.network` to `project_authelia` will resolve the warning, because compose will always prefix the network with the stack name.

Reply:
The other thing is, (I haven't done it yet) by adding the `name: {}` key to the top level `networks:` section we can avoid issues with docker compose appending the project name to the network.

```yaml
networks:
  proxy:
    external: true
    name: proxy
  authelia:
     name: authelia
```